pkg/cmd/server/api/validation/master.go

```diff
@@ -61,6 +61,13 @@ func ValidateMasterConfig(config *api.MasterConfig) ValidationResults {
 		validationResults.AddErrors(urlErrs...)
 	}
 
+	switch {
+	case config.ControllerLeaseTTL > 300:
+		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
+	case config.ControllerLeaseTTL > 0 && config.ControllerLeaseTTL < 10:
+		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
+	}
```

Comment:
Not sure what you mean, <10 isn't going to work and >300 isn't useful.

On Fri, Aug 14, 2015 at 3:59 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/api/validation/master.go
> <https://github.com/openshift/origin/pull/4137#discussion_r37113502>:
>
> > @@ -61,6 +61,13 @@ func ValidateMasterConfig(config *api.MasterConfig) ValidationResults {
> >  		validationResults.AddErrors(urlErrs...)
> >  	}
> >
> > +	switch {
> > +	case config.ControllerLeaseTTL > 300:
> > +		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
> > +	case config.ControllerLeaseTTL > 0 && config.ControllerLeaseTTL < 10:
> > +		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
> > +	}
>
> check for something other than -1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4137/files#r37113502>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift